src-tauri/src/lib.rs

```diff
@@ -39,12 +54,35 @@ fn saveas(handle: tauri::AppHandle) {
 }
 
 fn openfile(handle: &tauri::AppHandle, path: &Path) {
-    let h = handle.clone();
-    if let Ok(mut f) = File::open(path) {
-        let mut buffer = String::new();
-        f.read_to_string(&mut buffer).unwrap();
-        h.emit("open", (path.to_str().unwrap(), buffer)).unwrap();
-    }
+    let buffer = if path.extension().is_some_and(|ext| ext == "typ") {
+        let Ok(entry) = VERSE.entry_state().try_select_path_in_workspace(path) else {
+            eprintln!("not selected path {:?}", VERSE.entry_state());
+            return;
+        };
+        let world = VERSE.snapshot_with(Some(TaskInputs {
+            entry,
+            ..TaskInputs::default()
+        }));
+        if world.main_id().is_none() {
+            eprintln!("no main id");
+            return;
+        }
+
+        let Some(mut doc) = tyx_tiptap_typst::convert(Arc::new(world)) else {
+            return;
+        };
+        doc.filename = path.to_str().unwrap().to_string();
```

Comment:
This is unnecessary as it gets overridden in TypeScript in the onOpen function:

```typescript
openDocuments.push({ ...JSON.parse(content), filename })
```

On the other hand, we will need to change that function to rename the filename to end in `.tyx` and also not override the `.typ` file, perhaps something like `basename (Imported).tyx`